File: be/test/runtime/plugin/cloud_plugin_config_provider_test.cpp
// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

#include "runtime/plugin/cloud_plugin_config_provider.h"

#include <gtest/gtest-message.h>
#include <gtest/gtest-test-part.h>

#include <memory>
#include <string>

#include "common/status.h"
#include "gtest/gtest.h"
#include "runtime/plugin/s3_plugin_downloader.h"

namespace doris {

class CloudPluginConfigProviderTest : public ::testing::Test {
protected:
};

TEST_F(CloudPluginConfigProviderTest, TestGetCloudS3ConfigMethodExists) {
    // Test that get_cloud_s3_config method exists and compiles
    // We don't actually call it to avoid storage engine dependencies
    std::unique_ptr<S3PluginDownloader::S3Config> s3_config;
    // Just verify method signature exists by taking its address
    auto method_ptr = &CloudPluginConfigProvider::get_cloud_s3_config;
    EXPECT_TRUE(method_ptr != nullptr);
}

TEST_F(CloudPluginConfigProviderTest, TestGetCloudInstanceIdMethodExists) {
    // Test that get_cloud_instance_id method exists and compiles
    // We don't actually call it to avoid storage engine dependencies
    std::string instance_id;
    // Just verify method signature exists by taking its address
    auto method_ptr = &CloudPluginConfigProvider::get_cloud_instance_id;
    EXPECT_TRUE(method_ptr != nullptr);
}

TEST_F(CloudPluginConfigProviderTest, TestBasicClassStructure) {
    // Test that Status and required types are available
    Status test_status = Status::OK();
    EXPECT_TRUE(test_status.ok());

    // Test that S3Config type is accessible
    std::unique_ptr<S3PluginDownloader::S3Config> config;
    EXPECT_TRUE(config == nullptr);
}

} // namespace doris

File: fe/fe-core/src/test/java/org/apache/doris/common/plugin/CloudPluginConfigProviderTest.java
// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

package org.apache.doris.common.plugin;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class CloudPluginConfigProviderTest {

    @Test
    public void testGetCloudS3ConfigWithoutCloudEnvironment() {
        // Test getting S3 config without actual cloud environment
        // This should throw RuntimeException since we're not in cloud mode
        Assertions.assertThrows(RuntimeException.class, CloudPluginConfigProvider::getCloudS3Config);
    }

    @Test
    public void testGetCloudInstanceIdWithoutCloudEnvironment() {
        // Test getting cloud instance ID without actual cloud environment
        // This should throw RuntimeException since we're not in cloud mode
        Assertions.assertThrows(RuntimeException.class, CloudPluginConfigProvider::getCloudInstanceId);
    }

    @Test
    public void testCloudPluginConfigProviderClassExists() {
        // Basic test that the class exists and methods are accessible
        Assertions.assertNotNull(CloudPluginConfigProvider.class);

        // Verify methods exist by checking they can be called (will throw exceptions due to no cloud env)
        try {
            CloudPluginConfigProvider.getCloudS3Config();
            Assertions.fail("Should have thrown RuntimeException");
        } catch (RuntimeException e) {
            // Expected - no cloud environment available in test
            Assertions.assertNotNull(e.getMessage());
        }

        try {
            CloudPluginConfigProvider.getCloudInstanceId();
            Assertions.fail("Should have thrown RuntimeException");
        } catch (RuntimeException e) {
            // Expected - no cloud environment available in test
            Assertions.assertNotNull(e.getMessage());
        }
    }
}
